### DISCUSSION TITLE
What is the status and roadmap of this fork?

### DISCUSSION THREAD

**Discussion by udif** [firstof9/tplink-ess-lib#39] [General]
I found this fork while looking for a tool/library to parse the web management consoles of the TPLink smart switch series.
I was actually hoping to find a generic tool that supports many more switches (Zyxel, Cisco SG200, etc.), but this is a good start.
It seems that your fork is actively being worked on, but you threw away the command line tool part and instead you are building a library out of it, but no examples or documentations are given.

Is your fork the right place for me?

**Comment by lyricnz** [comment]
This library isn't generic at all - it's intended for TP-Link Easy Smart Switches only, and any generic tool could feel free to use this to talk to those switches, and other libraries to talk to other things.  The library is used, for example, in our Home Assistant integration at https://github.com/lyricnz/tplink_ess

This code started out in https://github.com/philippechataignon/smrt - but that hasn't had a non-typo commit in ~2 years, and there was no response to issues raised, so I'd consider it abandoned.

**Comment by lyricnz** [comment]
I created an issue against this project to add documentation and examples https://github.com/firstof9/tplink-ess-lib/issues/40
